components/ConvertForm/ManageConversion.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useCallback, useEffect } from 'react'
+import {
+  useOpenOrder,
+  useClaimOrder,
+  useApprove,
+  useAllowance,
+} from 'lib/web3-contracts'
+import { bigNum } from 'lib/utils'
+import ConvertSteps from 'components/ConvertSteps/ConvertSteps'
+import processing from './assets/loader.gif'
+
+function ManageConversion({ toAnj, fromAmount, toAmount, handleReturnHome }) {
+  const openOrder = useOpenOrder()
+  const claimOrder = useClaimOrder()
+  const changeAllowance = useApprove()
+  const getAllowance = useAllowance()
+  const [conversionSteps, setConversionSteps] = useState()
+
+  // Interacting with the bonding curve involves 2, 3 or 4 transactions (depending on the direction and state of allowance):
+  // 1. Reset approval (If we're converting ANT -> ANJ, an allowance was previously set but abandoned)
+  // 2. Raise approval (If we're converting ANT -> ANJ, the current allowance is not high enough)
+  // 3. Open a buy order
+  // 4. Claim the order
+  const createConvertSteps = useCallback(async () => {
+    let buyOrderHash
+    let steps = []
+
+    // We first check for allowance if the direction is correct
+    if (toAnj) {
+      const allowance = await getAllowance()
+
+      // and if we need more, add a step to ask to approve more
+      if (allowance.lt(bigNum(fromAmount))) {
+        steps = [
+          [
+            'Raise approval',
+            {
+              createTx: () => changeAllowance(fromAmount),
+            },
+          ],
+          ...steps,
+        ]
+      }
+
+      // Then there's the case when an user has an allowance set to the market maker contract
+      // but wants to convert even more tokens this time. When dealing with this case,
+      // we want to reset the allowance back to zero, and then raise it.
```

Comment:
```suggestion
      // we want to first prepend a transaction to reset the allowance back to zero (before raising it in the next transaction from above).
```

I might also do this in the condition above since it's related to that case:

```js
if (allowance.lt(bigNum(fromAmount))) {
  steps = [['Raise approval', ...]

  if (!allowance.isZero()) {
    steps = [['Reset approval', ...], ...steps]
  }
}
```

Reply:
👍